python/paddle/fluid/initializer.py

```diff
@@ -147,6 +165,7 @@ def __call__(self, var, block):
         """
         assert isinstance(var, framework.Variable)
         assert isinstance(block, framework.Block)
+        init_op = _is_inited_by(block, var, ['fill_constant'])
```

Comment:
I think it's better to check and avoid duplicated initialization in `create_parameter`? If the parameter already exists and shared, don't create `Initializer` (and check if the new initializer is the same as the old one)